### PR TITLE
Fix broadcast on master and some deprecations

### PR DESCRIPTION
**Pull request by andreasnoack**

### CODE REVIEW

**Comment by nalimilan** [comment]
Thanks, these failures are really annoying for DataFrames. Though you'll need to check `VERSION` for `broadcast_indices` too. See https://github.com/JuliaStats/NullableArrays.jl/pull/164.

**Comment by andreasnoack** [comment]
Done!

**Comment by andreasnoack** [comment]
I'll do that.

File: test/reducedim.jl [hunk 1]
```diff
@@ -127,7 +127,11 @@ for Areduc in (DataArray(rand(3, 4, 5, 6)),
             (1, 2, 3), (1, 3, 4), (2, 3, 4), (1, 2, 3, 4)]
             # println("region = $region, skipna = $skipna")
 
-            outputs = Any[DataArray(fill(NaN, Base.reduced_dims(size(Areduc), region)))]
+            if VERSION < v"0.6-"
```
Comment (tkelman) [inline test/reducedim.jl]: please be more specific about version cutoffs like this

File: REQUIRE [hunk 2]
```diff
@@ -1,4 +1,4 @@
 julia 0.4
 StatsBase 0.3
-Compat 0.8.6
+Compat 0.9.3
```
Comment (tkelman) [inline REQUIRE]: what needs 0.9.3? dot broadcast rewriting has been implemented for longer than that

File: test/reducedim.jl (hunk 1 above)
Comment (andreasnoack) [inline test/reducedim.jl]: I missed this one. The other conditions in the PR are precise.

File: REQUIRE (hunk 2 above)
Comment (andreasnoack) [inline REQUIRE]: Nothing that I'm aware of but I thought I'd better bump it to match the version used for the testing.

**Pull request merged**